Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs

```diff
@@ -245,7 +245,7 @@ public Task<BoolResult> RequestCopyFileAsync(OperationContext context, ContentHa
             return _proactiveCopyIoGate.GatedOperationAsync(ts =>
                 {
                     var cts = new CancellationTokenSource();

```

Comment:
we need to wrap cts into a using block (or use using var cts).

This is very important, because currently, we'll trigger the timer even when the operation finishes successfully.

THis can be pretty bad if we have hundreds of operations like this every let's say a minute.

Reply:
but one very important point here. The method should be made async and we should do 'return await context.PerformOperationAsync()'

to avoid early release of cancellation token source.

---
In reply to: [342741588](https://github.com/microsoft/BuildXL/pull/1141#discussion_r342741588) [](ancestors = 342741588)